Refer to this link for build results (access rights to CI server needed): 
http://bgate.mellanox.com/jenkins/job/gh-libibprof-pr/117/<h2>Build result: FAILURE</span></h2>[...truncated 90 lines...]   at hudson.model.Run.execute(Run.java:1786)  at hudson.matrix.MatrixBuild.run(MatrixBuild.java:306)  at hudson.model.ResourceController.execute(ResourceController.java:98)  at hudson.model.Executor.run(Executor.java:374)Caused by: java.io.FileNotFoundException: https://api.github.com/repos/mellanox-hpc/libibprof/statuses/775f8608bedba2d74e175f08f5a3632d7c018f99  at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)    at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57) at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) at java.lang.reflect.Constructor.newInstance(Constructor.java:526)  at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1675)   at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1673)   at java.security.AccessController.doPrivileged(Native Method)   at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1671) at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1244)  at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)    at org.kohsuke.github.Requester.parse(Requester.java:451)   at org.kohsuke.github.Requester._to(Requester.java:224) ... 13 moreCaused by: java.io.FileNotFoundException: https://api.github.com/repos/mellanox-hpc/libibprof/statuses/775f8608bedba2d74e175f08f5a3632d7c018f99  at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1624)  at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:468)   at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338)   at org.kohsuke.github.Requester.parse(Requester.java:447)   ... 14 more

**Build Log**
_last 50 lines_

```
[...truncated 65 lines...]
Cleaning workspace
 > /usr/bin/git rev-parse --verify HEAD # timeout=60
Resetting working tree
 > /usr/bin/git reset --hard # timeout=60
 > /usr/bin/git clean -fdx # timeout=60
No emails were triggered.
Triggering gh-libibprof-pr » master
gh-libibprof-pr » master completed with result FAILURE
Email was triggered for: Failure - Any
Sending email for trigger: Failure - Any
An attempt to send an e-mail to empty list of recipients, ignored.
Setting status of 775f8608bedba2d74e175f08f5a3632d7c018f99 to FAILURE with url http://bgate.mellanox.com:8888/jenkins/job/gh-libibprof-pr/117/ and message: Merged build finished.
FileNotFoundException means that the credentials Jenkins is using is probably wrong. Or that something is really wrong with github.
java.io.FileNotFoundException: {"message":"Not Found","documentation_url":"https://developer.github.com/v3"}
    at org.kohsuke.github.Requester.handleApiError(Requester.java:494)
    at org.kohsuke.github.Requester._to(Requester.java:245)
    at org.kohsuke.github.Requester.to(Requester.java:191)
    at org.kohsuke.github.GHRepository.createCommitStatus(GHRepository.java:780)
    at org.kohsuke.github.GHRepository.createCommitStatus(GHRepository.java:792)
    at org.jenkinsci.plugins.ghprb.GhprbRepository.createCommitStatus(GhprbRepository.java:140)
    at org.jenkinsci.plugins.ghprb.GhprbRepository.createCommitStatus(GhprbRepository.java:122)
    at org.jenkinsci.plugins.ghprb.GhprbBuilds.onCompleted(GhprbBuilds.java:118)
    at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onCompleted(GhprbBuildListener.java:27)
    at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onCompleted(GhprbBuildListener.java:12)
    at hudson.model.listeners.RunListener.fireCompleted(RunListener.java:199)
    at hudson.model.Run.execute(Run.java:1786)
    at hudson.matrix.MatrixBuild.run(MatrixBuild.java:306)
    at hudson.model.ResourceController.execute(ResourceController.java:98)
    at hudson.model.Executor.run(Executor.java:374)
Caused by: java.io.FileNotFoundException: https://api.github.com/repos/mellanox-hpc/libibprof/statuses/775f8608bedba2d74e175f08f5a3632d7c018f99
    at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
    at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
    at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
    at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1675)
    at sun.net.www.protocol.http.HttpURLConnection$6.run(HttpURLConnection.java:1673)
    at java.security.AccessController.doPrivileged(Native Method)
    at sun.net.www.protocol.http.HttpURLConnection.getChainedException(HttpURLConnection.java:1671)
    at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1244)
    at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
    at org.kohsuke.github.Requester.parse(Requester.java:451)
    at org.kohsuke.github.Requester._to(Requester.java:224)
    ... 13 more
Caused by: java.io.FileNotFoundException: https://api.github.com/repos/mellanox-hpc/libibprof/statuses/775f8608bedba2d74e175f08f5a3632d7c018f99
    at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1624)
    at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:468)
    at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338)
    at org.kohsuke.github.Requester.parse(Requester.java:447)
    ... 14 more
```

Test FAILed.

--- a/configure.ac
+++ b/configure.ac
@@ -46,7 +46,7 @@ CPPFLAGS="$CPPFLAGS -D_GNU_SOURCE -D__LINUX__ -D__MODULE_VERSION=$VERSION"
 #
 AC_ARG_WITH([ofed],
             [AC_HELP_STRING([--with-ofed(=DIR)],
-                            [Build OpenFabrics support, adding DIR/include, DIR/lib, and DIR/lib64 to the search path for headers and libraries])],
+                            [IBPROF: build OpenFabrics support, adding DIR/include, DIR/lib, and DIR/lib64 to the search path for headers and libraries])],
             [],
             [with_ofed=/usr]
             )
@@ -195,6 +195,55 @@ if test $MOFED_VERSION -gt 0; then
 	AC_MSG_NOTICE([Current VERBS API version : $MOFED_VERSION])
 else
 	AC_MSG_NOTICE([Current VERBS API version : legacy])
+fi
+
+# Check if the compiler has support for visibility, like some
+# versions of gcc, icc Sun Studio cc.
+AC_ARG_ENABLE(visibility, 
+    AC_HELP_STRING([--enable-visibility],
+        [IBPROF: enable visibility feature of certain compilers/linkers (default: enabled)]))
+
+ac_visibility_define=0
+ac_msg="whether to enable symbol visibility"
+
+if test "$ac_enable_visibility" = "no"; then
+    AC_MSG_CHECKING([$ac_msg])
+    AC_MSG_RESULT([no (disabled)]) 
+else
+    CFLAGS_orig=$CFLAGS
+
+    # Check using -fvisibility=hidden
+    ac_add=-fvisibility=hidden
+    CFLAGS="$ac_add -Werror"
+
+    AC_MSG_CHECKING([if $CC supports $ac_add])
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([[
+        #include <stdio.h>
+        __attribute__((visibility("default"))) int foo;
+        ]],[[fprintf(stderr, "Hello, world\n");]])],
+        [AS_IF([test -s conftest.err],
+               [$GREP -iq visibility conftest.err
+                # If we find "visibility" in the stderr, then
+                # assume it doesn't work
+                AS_IF([test "$?" = "0"], [ac_add=])])
+        ], [ac_add=])
+    AS_IF([test "$ac_add" = ""],
+          [AC_MSG_RESULT([no])],
+          [AC_MSG_RESULT([yes])])
+
+    CFLAGS=$CFLAGS_orig
+
+    if test "$ac_add" != "" ; then
+        AC_MSG_CHECKING([$ac_msg])
+        AC_MSG_RESULT([yes (via $ac_add)])
+        CFLAGS="$CFLAGS -fvisibility=hidden -DHAVE_VISIBILITY"
+    elif test "$enable_visibility" = "yes"; then
+        AC_MSG_ERROR([Symbol visibility support requested but compiler does not seem to support it.  Aborting])
+    else
+        AC_MSG_CHECKING([$ac_msg])
+        AC_MSG_RESULT([no (unsupported)])
+    fi
+    unset ac_add
 fi
 
 RPM_RELEASE=1

--- a/src/api/ibprof_api.c
+++ b/src/api/ibprof_api.c
@@ -59,7 +59,7 @@ static IBPROF_MODULE_OBJECT *__ibprof_modules[] = {
 /*
  * It is the core of profiling facility
  */
-static IBPROF_OBJECT *ibprof_obj;	/* Verify a pointer to this object with NULL to check ACTIVE/CLOSE */
+static IBPROF_OBJECT *ibprof_obj = NULL;	/* Verify a pointer to this object with NULL to check ACTIVE/CLOSE */
 pthread_once_t ibprof_initialized = PTHREAD_ONCE_INIT;
 
 
@@ -334,10 +334,10 @@ void __attribute__((constructor)) __ibprof_init(void)
 				}
 
 				if (temp_ibprof_obj->hash_obj)
-					ibprof_hash_destroy(ibprof_obj->hash_obj);
+					ibprof_hash_destroy(temp_ibprof_obj->hash_obj);
 
 				if (temp_ibprof_obj->task_obj)
-					ibprof_task_destroy(ibprof_obj->task_obj);
+					ibprof_task_destroy(temp_ibprof_obj->task_obj);
 
 				sys_free(temp_ibprof_obj);
 			}
@@ -404,7 +404,7 @@ void __attribute__((destructor)) __ibprof_exit(void)
 			sys_fclose(ibprof_dump_file);
 			if (!sys_fstat(filename, &statbuf))
 				if (!statbuf.st_size)
-					  sys_fremove(filename);
+					  ret = sys_fremove(filename);
 		}
 	        sys_free(filename);
 	}

--- a/src/api/ibprof_api.h
+++ b/src/api/ibprof_api.h
@@ -22,6 +22,10 @@
 extern "C" {
 #endif
 
+#if defined(HAVE_VISIBILITY)
+#pragma GCC visibility push(default)
+#endif
+
 /**
  * @enum
  * @brief List of supported module types.
@@ -110,6 +114,9 @@ void ibprof_interval_end(int callid);
  ***************************************************************************/
 void ibprof_dump(void);
 
+#if defined(HAVE_VISIBILITY)
+#pragma GCC visibility pop
+#endif
 
 #ifdef __cplusplus
 }

--- a/src/core/hcol/ibprof_hcol.c
+++ b/src/core/hcol/ibprof_hcol.c
@@ -92,7 +92,13 @@ static struct module_context_t {
 DECLARE_OPTION_FUNCTIONS_PROTOTYPED(type) \
 DECLARE_OPTION_STRUCT(type)
 
+#if defined(HAVE_VISIBILITY)
+#pragma GCC visibility push(default)
+#endif
 DECLARE_OPTION_FUNCTIONS_PROTOTYPED( )
+#if defined(HAVE_VISIBILITY)
+#pragma GCC visibility pop
+#endif
 
 #define PREFIX_NONE(x) NONE##x,
 DECLARE_OPTION(NONE)

--- a/src/core/ibprof_conf.c
+++ b/src/core/ibprof_conf.c
@@ -136,6 +136,7 @@ static void _ibprof_conf_dump_file(char *str)
 	char buf[1024];
 	char *dest = buf;
 	int dest_len = 0;
+        char *tmp = NULL;
 
 	while (pattern && ((max_len - dest_len) > 1)) {
 		char c = *pattern++;
@@ -160,11 +161,13 @@ static void _ibprof_conf_dump_file(char *str)
 			break;
 
 		case 'H': /* add Host name */
+                        tmp = sys_host();
 			ret =
 					sys_snprintf_safe(
 							(dest + dest_len),
 							(max_len - dest_len),
-							"%s", sys_host());
+							"%s", tmp);
+			sys_free(tmp);
 			break;
 
 		case 'T': /* add Task ID */

--- a/src/core/ibv/ibprof_ibv.c
+++ b/src/core/ibv/ibprof_ibv.c
@@ -853,7 +853,13 @@ DECLARE_OPTION_FUNCTIONS_PROTOTYPED(type) \
 DECLARE_OPTION_FUNCTIONS_INLINE(type) \
 DECLARE_OPTION_STRUCT(type)
 
+#if defined(HAVE_VISIBILITY)
+#pragma GCC visibility push(default)
+#endif
 DECLARE_OPTION_FUNCTIONS_PROTOTYPED( )
+#if defined(HAVE_VISIBILITY)
+#pragma GCC visibility pop
+#endif
 
 #define PREFIX_NONE(x) NONE##x,
 DECLARE_OPTION(NONE)

--- a/src/core/mxm/ibprof_mxm.c
+++ b/src/core/mxm/ibprof_mxm.c
@@ -163,7 +163,13 @@ static struct module_context_t {
 DECLARE_OPTION_FUNCTIONS_PROTOTYPED(type) \
 DECLARE_OPTION_STRUCT(type)
 
+#if defined(HAVE_VISIBILITY)
+#pragma GCC visibility push(default)
+#endif
 DECLARE_OPTION_FUNCTIONS_PROTOTYPED( )
+#if defined(HAVE_VISIBILITY)
+#pragma GCC visibility pop
+#endif
 
 #define PREFIX_NONE(x) NONE##x,
 DECLARE_OPTION(NONE)

